phpBB/phpbb/db/migration/data/v33x/bot_update_vol2.php

```diff
@@ -0,0 +1,99 @@
+<?php
+/**
+ *
+ * This file is part of the phpBB Forum Software package.
+ *
+ * @copyright (c) phpBB Limited <https://www.phpbb.com>
+ * @license GNU General Public License, version 2 (GPL-2.0)
+ *
+ * For full copyright and license information, please see
+ * the docs/CREDITS.txt file.
+ *
+ */
+
+namespace phpbb\db\migration\data\v33x;
+
+class bot_update_vol2 extends \phpbb\db\migration\migration
+{
+	static public function depends_on()
+	{
+		return ['\phpbb\db\migration\data\v33x\v334'];
+	}
+
+	public function update_data()
+	{
+		return [
+			['custom', [[$this, 'add_bots']]],
+		];
+	}
+
+	public function add_bots()
+	{
+		$bots = [
+			'Ahrefs [Bot]' => 'AhrefsBot/',
+			'Amazon [Bot]' => 'Amazonbot/',
+			'Semrush [Bot]' => 'SemrushBot/'
+		];
+
+		foreach ($bots as $bot_name => $bot_agent)
+		{
+			$bot_name_clean = utf8_clean_string($bot_name);
+
+			$sql = 'SELECT user_id
+				FROM ' . $this->table_prefix . 'users
+				WHERE ' . $this->db->sql_build_array('SELECT', ['username_clean' => $bot_name_clean]);
+			$result = $this->db->sql_query($sql);
+			$bot_exists = (bool) $this->db->sql_fetchfield('user_id');
+			$this->db->sql_freeresult($result);
+
+			if ($bot_exists)
+			{
+				continue;
+			}
+
+			$sql = 'SELECT group_id, group_colour
+				FROM ' . $this->table_prefix . 'groups
+				WHERE ' . $this->db->sql_build_array('SELECT', ['group_name' => 'BOTS']);
+			$result = $this->db->sql_query($sql);
+			$group_row = $this->db->sql_fetchrow($result);
+			$this->db->sql_freeresult($result);
+
+			// Default fallback, should never get here
+			if (!$group_row)
+			{
+				$group_row['group_id'] = 6;
+				$group_row['group_colour'] = '9E8DA7';
+			}
```

Comment:
```suggestion
			if (!count($group_row))
			{
				$sql = 'SELECT group_id, group_colour
					FROM ' . $this->table_prefix . 'groups
					WHERE ' . $this->db->sql_build_array('SELECT', ['group_name' => 'BOTS']);
				$result = $this->db->sql_query($sql);
				$group_row = $this->db->sql_fetchrow($result);
				$this->db->sql_freeresult($result);

				// Default fallback, should never get here
				if (!count($group_row))
				{
					$group_row['group_id'] = 6;
					$group_row['group_colour'] = '9E8DA7';
				}
			}
```

Reply:
With this and the above change we won't have to re-run the same query three times.